GitUI/UserControls/SortDiffListContextMenuItem.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+using GitCommands;
+using GitUI.Properties;
+using ResourceManager;
+
+namespace GitUI.UserControls
+{
+    public class SortDiffListContextMenuItem : ToolStripMenuItem
+    {
+        private readonly TranslationString _sortByText = new TranslationString("&Sort by...");
+        private readonly TranslationString _filePathSortText = new TranslationString("File &Path");
+        private readonly TranslationString _fileExtensionSortText = new TranslationString("File &Extension");
+        private readonly TranslationString _fileStatusSortText = new TranslationString("File &Status");
+        private readonly IDiffListSortService _sortService;
+        private readonly ToolStripMenuItem _filePathSortItem;
+        private readonly ToolStripMenuItem _fileExtensionSortItem;
+        private readonly ToolStripMenuItem _fileStatusSortItem;
+
+        public SortDiffListContextMenuItem(IDiffListSortService sortService)
+        {
+            _sortService = sortService ?? throw new ArgumentNullException(nameof(sortService));
+            Image = Images.SortBy;
+            Text = _sortByText.Text;
+
+            _filePathSortItem = new ToolStripMenuItem()
+            {
+                Text = _filePathSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FilePath
+            };
+
+            _fileExtensionSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileExtensionSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileExtension
+            };
+
+            _fileStatusSortItem = new ToolStripMenuItem()
+            {
+                Text = _fileStatusSortText.Text,
+                ShowShortcutKeys = true,
+                Image = null,
+                Tag = DiffListSortType.FileStatus
+            };
+
+            foreach (var item in AllItems())
+            {
+                item.Click += Item_Click;
+                DropDownItems.Add(item);
+            }
+
+            RequerySortingMethod();
+        }
+
+        private ToolStripMenuItem[] AllItems()
+        {
+            return new[] { _filePathSortItem, _fileExtensionSortItem, _fileStatusSortItem, };
+        }
+
+        public void RequerySortingMethod()
+        {
+            var currentSort = _sortService.DiffListSorting;
+            foreach (var item in AllItems())
+            {
+                item.Checked = currentSort.Equals(item.Tag);
+            }
+        }
```

Comment:
Can't we do something like (pseudo code):
```
this.OnOpening += (s, e) => {
            var currentSort = _sortService.DiffListSorting;
            foreach (var item in AllItems())
            {
                item.Checked = currentSort.Equals(item.Tag);
            }
}
```

Reply:
Seams reasonable. I'll try that out. I don't know winforms classes much so I didn't think of that.

Reply:
Will Do.